api/helpers/gamestate/moves/resolve/execute.js

```diff
@@ -0,0 +1,66 @@
+const GamePhase = require('../../../../../utils/GamePhase.json');
+const MoveType = require('../../../../../utils/MoveType.json');
+
+module.exports = {
+  friendlyName: 'Resolve One-Off',
+
+  description: 'Returns new GameState resulting from requested resolve move',
+
+  inputs: {
+    currentState: {
+      type: 'ref',
+      description: 'The latest GameState before the requesting player resolves the pending one-off and counters',
+      required: true,
+    },
+    /**
+     * @param { Object } requestedMove - The move being requested. 
+     * Specifies that the move is to resolve
+     * @param { MoveType.RESOLVE } requestedMove.moveType
+     */
+    requestedMove: {
+      type: 'ref',
+      description: 'The move being requested. Specifies which player is asking to play a one-off ',
+    },
+    playedBy: {
+      type: 'number',
+      description: 'Player number of player requesting move.',
+    },
+  },
+  sync: true, // synchronous helper
+  fn: ({ currentState, requestedMove, playedBy }, exits) => {
+    let result = _.cloneDeep(currentState);
+
+    const { oneOff } = result;
+
+    // Move oneOff + twos into scrap and increment turn
+    result.scrap.push(result.oneOff);
+    result.scrap.push(...result.twos);
+    result = {
+      ...result,
+      ...requestedMove,
+      phase: GamePhase.MAIN,
+      playedBy,
+      resolved: oneOff,
+      oneOff: null,
+      twos: [],
+      turn: result.turn + 1,
+    };
```

Comment:
This is all the generic handling shared by all the one-offs that fully resolve without needing additional player input. It will be more complicated for threes, fours, fives, and sevens, which then require the player to do additional stuff. It's unclear to me atm whether to have those cases consume `currentState` instead of `result` to effectively ignore these changes and recreate their own, or whether to have them roll back the unnecessary changes by explicitly setting these values e.g. `turn` differently. I lean towards just having them manipulate `currentState` instead of `result``.